src/test/java/uk/gov/hmcts/reform/draftstore/utils/RetryTest.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.draftstore.utils;
+
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class RetryTest {
+
+    @Test
+    public void should_return_result_of_calling_an_action_with_first_param_if_it_succeeded() throws Exception {
+        int result = Retry.with(asList(1, 2, 3), n -> failOnZero(n));
+        assertThat(result).isEqualTo(1);
+    }
+
+    @Test
+    public void should_return_result_of_calling_an_action_with_second_param_if_first_one_failed() throws Exception {
+        int result = Retry.with(asList(0, 24), n -> failOnZero(n));
+        assertThat(result).isEqualTo(24);
+    }
+
+    @Test
+    public void should_throw_an_exception_when_action_fails_for_all_params() throws Exception {
+        assertThatThrownBy(
+            () -> Retry.with(asList(0, 0, 0), n -> failOnZero(n))
+        ).isInstanceOf(RuntimeException.class);
```

Comment:
It would be ideal to check if it's the exception thrown by the function wrapped by this Retry. Or maybe the tool should throw a higher-level exception, with this exception as the cause?